src/main/java/com/hotels/bdp/circustrain/bigquery/partition/HivePartitionGenerator.java

```diff
@@ -130,32 +130,29 @@ public HivePartitionGenerator(
     List<GeneratePartitionTask> tasks = new ArrayList<>();
     for (FieldValueList row : rows) {
       tasks
-          .add(new GeneratePartitionTask(sourceDBName, sourceTableName, partitionKey, partitionKeyType, tableBucket,
+          .add(new GeneratePartitionTask(sourceTableAsHive, partitionKey, partitionKeyType, tableBucket,
               tableFolder, row));
     }
     return tasks;
   }
 
   private class GeneratePartitionTask implements Callable<Optional<Partition>> {
 
-    private final String sourceDBName;
-    private final String sourceTableName;
+    private final Table sourceTableAsHive;
     private final String partitionKey;
     private final String partitionKeyType;
     private final String tableBucket;
     private final String tableFolder;
     private final FieldValueList row;
 
     private GeneratePartitionTask(
-        String sourceDBName,
-        String sourceTableName,
+        Table sourceTableAsHive,
```

Comment:
whats the meaning of `AsHive`?

Reply:
ah seen the thread above, but i think you've missed it here?

Reply:
The table to replicate is a BigQuery table and a hive table is created based on the schema from that. So you will have the source table, and then the hive version of the source table. It was already named as `sourceTableAsHive` but Adrians suggested I should change it since its a bit confusing.

Reply:
yeah I missed one! Thanks for pointing it out